test/backend-unit-tests/cmake/environment.test.ts

```diff
@@ -0,0 +1,60 @@
+import { expect } from 'chai';
+import { EnvironmentVariablesUtils } from '@cmt/environmentVariables';
+
+suite('[Environment]', async () => {
+    test('Environment variable to `preserve/non-preserve null` `win/non-win`', () => {
+        const envA = {
+            A: 'x',
+            B: null
+        };
+        const envB = {
+            a: 'T',
+            u: 'BBQ'
+        };
+        const resultA = EnvironmentVariablesUtils.mergeImpl(false, false, envA, undefined, envB);
+        expect(resultA).to.deep.equal({A: 'x', a: 'T', u: 'BBQ'});
+        const resultB = EnvironmentVariablesUtils.mergeImpl(true, false, envA, undefined, envB);
+        expect(resultB).to.deep.equal({A: 'x', B: null, a: 'T', u: 'BBQ'});
+        const resultC = EnvironmentVariablesUtils.mergeImpl(false, true, envA, undefined, envB);
+        expect(resultC).to.deep.equal({A: 'T', u: 'BBQ'});
+        const resultD = EnvironmentVariablesUtils.mergeImpl(true, true, envA, undefined, envB);
+        expect(resultD).to.deep.equal({A: 'T', B: null, u: 'BBQ'});
```

Comment:
Could you also add a test that has a `null` in envB override a value from envA? That's the presets case that I want to ensure doesn't break.

Reply:
Added